underfs/s3a/src/main/java/alluxio/underfs/s3a/AlluxioS3Exception.java

```diff
@@ -31,12 +31,26 @@ public class AlluxioS3Exception extends AlluxioRuntimeException {
    * @return alluxio s3 exception
    */
   public static AlluxioS3Exception from(AmazonClientException cause) {
+    return fromWithErrorMessage(null, cause);
+  }
+
+  /**
+   * Converts an AmazonClientException with errormessage to a corresponding AlluxioS3Exception.
+   * @param errorMessage error message
+   * @param cause aws s3 exception
+   * @return alluxio s3 exception
+   */
+  public static AlluxioS3Exception fromWithErrorMessage(String errorMessage,
```

Comment:
```suggestion
  public static AlluxioS3Exception from(String errorMessage,
```

I'd prefer if we overload the Java method. You'll need to refactor the name where affected.

Reply:
done